compiler/src/dmd/typinf.d

```diff
@@ -52,6 +52,13 @@ extern (C++) void genTypeInfo(Expression e, const ref Loc loc, Type torig, Scope
                 .error(loc, "expression `%s` uses the GC and cannot be used with switch `-betterC`", e.toChars());
             else
                 .error(loc, "`TypeInfo` cannot be used with -betterC");
+
+            if (sc && sc.tinst)
+            {
+                global.params.verbose = true;
+                sc.tinst.printInstantiationTrace();
```

Comment:
Looks like that global param should really be passed to `printInstantiationTrace` as a parameter.

Reply:
Will fix this up later

Reply:
https://github.com/dlang/dmd/pull/15605#issuecomment-1742341481